addons/website/views/snippets/snippets.xml

```diff
@@ -20,6 +20,8 @@
                 </div>
             </div>
 
+            <input type="text" class="o_snipet_search_filter_input" placeholder="Search..." />
```

Comment:
I think you can move your XML and JS logic in the web_editor app. This is not something specific to website only, we will want that snippet search for all editors (for example, for mass mailing too).